### DISCUSSION TITLE
Conditions with AND and OR

### DISCUSSION THREAD

**Discussion by Dumpstar56** [directus/directus#14679] [Archive]
I am having trouble creating a condition for this scenario. I hope someone can tell me what I am doing wrong.
I want a field or divider to be hidden until there is data entered into either one or both of two text fields, or a boolean is marked true. I have created this condition:
![condition issue](https://user-images.githubusercontent.com/94397183/181094551-dd2b1057-ec4c-4f5f-a006-2874afd43c03.jpg)
There are no other conditions on this field. It is not working as intended. 
- If data is entered into one text field only, or if the Boolean is marked true, the field with the condition remains hidden. 
- If one text field has data and the boolean is marked true, the field is no longer hidden. 
- If both text fields have data, the field is no longer hidden. 
- If both text fields have data and the boolean is marked true, the field is hidden.
What am I doing wrong?

**Comment by rijkvanzanten** [comment]
> I want a field or divider to be hidden until there is data entered into either one or both of two text fields, or a boolean is marked true

I think you can drop drop the AND group and instead use `is not null` on the text values 🤔

**Comment by Dumpstar56** [comment]
Thank you! That was the inspiration to figure out the way to do it. A simple AND group with text fields are null and Boolean is not true did the trick. I should have thought of that. Still seems like some of the conditions do not behave as they should. Anyway. Thanks!
